lib/Net/Amazon/S3/Client/Object.pm

```diff
@@ -263,11 +263,13 @@ sub delete {
 
 sub initiate_multipart_upload {
     my $self = shift;
+    my %args = ref($_[0]) ? %{$_[0]} : @_;
```

Comment:
this module already has dependency to `Ref::Util`, will be safer to use it to detect hashref directly
```suggestion
    my %args = Ref::Util::is_plain_ref($_[0]) ? %{$_[0]} : @_;
```